mscore/albummanager.cpp

```diff
@@ -19,213 +24,621 @@
-        if (t) {
-            name
-                = QTextDocumentFragment::fromHtml(t->xmlText()).toPlainText().replace("&amp;","&").replace("&gt;",
-                                                                                                           ">").replace(
-                      "&lt;","<").replace("&quot;", "\"");
+    for (int i = 0; i < _items.size(); i++) {
+        for (int j = 0; j < _items.size(); j++) {
+            if (_items.at(j)->albumItem->score->title() != scoreList->item(j, 0)->text()) {
+                int h = scoreList->row(scoreList->findItems(_items.at(j)->albumItem->score->title(),
+                                                            Qt::MatchExactly).first());
+                std::swap(_items.at(j), _items.at(h));
+                break;
+            } else if (j == _items.size() - 1) {
+                goto end;
+            }
         }
-        name = name.simplified();
     }
-    if (name.isEmpty()) {
-        name = score->title();
+end:;
+    updateButtons();
+}
+
+//---------------------------------------------------------
+//   openSettingsDialog
+///     (Re)Open the settings dialog menu.
+//---------------------------------------------------------
+
+void AlbumManager::openSettingsDialog(bool throwaway)
+{
+    if (!settingsDialog) {
+        settingsDialog = new AlbumManagerDialog(this);
+        settingsDialog->start();
+    } else {
+        settingsDialog->start();
     }
-    return name;
 }
 
+
+//---------------------------------------------------------
+//   albumScores
+///     this is a leftover before the refactor
+//---------------------------------------------------------
+
+std::vector<AlbumItem*> AlbumManager::albumScores() const
+{
+    return _album->_albumItems;
+}
+
+////---------------------------------------------------------
+////   getScoreTitle
+/////     TODO: What does this do and why?
+////---------------------------------------------------------
+
+//static QString getScoreTitle(Score* score)
+//{
+//    QString name = score->metaTag("movementTitle");
+//    if (name.isEmpty()) {
+//        Text* t = score->getText(Tid::TITLE);
+//        if (t) {
+//            name = QTextDocumentFragment::fromHtml(t->xmlText()).toPlainText().replace("&amp;","&")
+//                    .replace("&gt;",">").replace("&lt;","<").replace("&quot;", "\"");
+//        }
+//        name = name.simplified();
+//    }
+//    if (name.isEmpty()) {
+//        name = score->title();
+//    }
+//    return name;
+//}
+
 //---------------------------------------------------------
 //   addClicked
+///     Add an existing score to the Album.\n
+///     Opens a dialog to select a Score from the filesystem.
+///     TODO: if the score is already opened openScore returns nullptr,
+///     so this does not work
 //---------------------------------------------------------
 
-void AlbumManager::addClicked()
+void AlbumManager::addClicked(bool throwaway)
```

Comment:
https://wiki.qt.io/New_Signal_Slot_Syntax

Essentially the new syntax does not support default values, so I have a bool that I don't need when clicking buttons.

Reply:
Ah, I see. The convention for unused variables tends to be to call them `_`, but you can stick with `throwaway` if you want.

Reply:
Creating parameters and not using them tiggers compiler warnings. You can avoid the warnings by using the parameter in a no-op statement (`Q_UNUSED` exists for this purpose):

```c++
// header
void addClicked(bool throwaway = false)

// .cpp
void AlbumManager::addClicked(bool throwaway)
{
    Q_UNUSED(throwaway);
    ...
}
```

Or by not giving the variable any kind of name at all:

```c++
// header
addClicked(bool = false);

// .cpp
void AlbumManager::addClicked(bool)
{
    ...
}
```
Either way, I would leave a comment in the `.ccp` to say something like *"Unused bool parameter is necessary to connect this function to QClass::function via the signal-slot mechanism"*. If you do give the parameter a name, you could consider giving it the same name as it has in the Qt function.

Reply:
C++ truly has the answer to every problem xD
I had completely forgotten about not naming parameters and didn't know that Q_UNUSED existed. I decided to use Q_UNUSED because I prefer explicitness. Thanks :-)